src/conditions/vrl.rs

```diff
@@ -7,6 +7,7 @@ use vrl::value::Value;
 
 use crate::config::LogNamespace;
 use crate::event::TargetEvents;
+use crate::sources::dnstap;
```

Comment:
hm this might need some refactoring

Reply:
Yeah, I didn't like it either. I first wanted to use `dnsmsg_parser` module here, but I wanted to have the exact format used in the source.

Maybe it would make sense to move the parsing part of it to a separate crate? Then the vrl function could maybe be a separate crate too (or part of that parsing crate).

Reply:
By crate, do you perhaps mean module or lib? Originally I wondered if we can move this to the VRL repo but that's even more work.

Reply:
➕ to moving the shared bits to a module in `lib/`.

Reply:
Yeah, I was just about to write that I meant moving to `lib` (similar to how there is `lib/vector-vrl/functions`).

The thing I am not completely sure about is, where to put the VRL function itself.

I was going to move the dnstap parsing to `lib/dnstap` or something like that. Now the question is, should the function be yet another separate module, a part of `lib/dnstap`, or something else?
Looking at what is available in the project so far, it seems that making it a part of `lib/dnstap` would probably make most sense (since that is how it is done for `enrichment` related functions).

Reply:
👍 I don't have strong opinions. `dnstap-parser` sounds fine to me. Having the VRL function there seems ok too so long as it doesn't create strange dependencies 🤔 E.g. we wouldn't want the `dnstap` source module to depend on VRL.

Reply:
I would leave dnstap source in the main crate (inside `src/sources/dnstap`) and just move its parser and possibly schema out to `lib`. I guess in that case it is okay to leave the function in that `lib` crate?

Reply:
I think having the function in the `dns-parser` crate might create a dependency between `sources::dnstap` and VRL. Maybe it could be behind a feature flag?

Reply:
Yeah, we could do that too. `vrl` feature flag, or something like that

Reply:
I just checked this, we can put the function behind `vrl` feature flag, but the dependency on `vrl` will still be there (it seems it has always been there, since `src/sources/dnstap/schema` depends on `vrl` types).

Reply:
> I would leave dnstap source in the main crate (inside src/sources/dnstap) and just move its parser and possibly schema out to lib. I guess in that case it is okay to leave the function in that lib crate?

This sounds good to me. I am not sure I am following the feature flag discussion, I don't think we need it. But we can discuss after we have an updated version to review.

Reply:
I have moved it, let me know if the current version makes sense.